### PR TITLE
New design and logic of notifications

### PR DESCRIPTION
**Pull request by iiLubos**
Notification

NotificationModel - class for the logic of showing notification
MMNotificationView - view for a list of notification
MMNotification - notification

Create a notification (showing 10s) - Notification{ nextId(), "Message", 10, NotificationType::Information, NotificationType::Waiting }
Add new notification - add(const QString &message, uint interval, NotificationType::MessageType type, NotificationType::IconType icon)
Remove specific notification - remove(uint id)

Removing notifications is possible with a close button or will disappear automatically 

In this PR, also refactored Input fields
Inputs
Samples:
```
MMInput {
    placeholderText: "Search"
    iconSource: Style.searchIcon
}
MMInput {
    text: "Calendar"
    iconSource: Style.calendarIcon
    warningMsg: "Would you like to be so kind and select a date please?"
}
MMPasswordInput {
    text: "Password"
    regexp: '(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[^A-Za-z0-9])(?=.{6,})'
    errorMsg: "Password must contain at least 6 characters\nMinimum 1 number, uppercase and lowercase letter and special character"
}
MMButtonInput {
    buttonText: "Send"
    placeholderText: "Write an informative message"
    onClicked: notificationModel.add(text, 60, NotificationType.Information, NotificationType.None)
}
```

### CODE REVIEW

**Comment by PeterPetrik** [comment]
nice!

Q: How it behaves when notification is spanning on multiple lines? Or is it forbidden?

**Comment by iiLubos** [comment]
> nice!
> 
> Q: How it behaves when notification is spanning on multiple lines? Or is it forbidden?

For now, only one line is supported.
The notification can be responsive, if it makes sense

**Comment by PeterPetrik** [comment]
If it is longer it could possibly do animation ? (like moving/scrolling text banners in TV ?)

**Comment by tomasMizera** [comment]
> If it is longer it could possibly do animation ? (like moving/scrolling text banners in TV ?)

I would leave this decision with Andrej :)

**Comment by iiLubos** [comment]
Notification is now responsible, max. 3 rows.

File: app/qmlV2/component/MMInput.qml [hunk 1]
```diff
@@ -15,14 +15,16 @@ import "../Style.js" as Style
 Item {
   id: control
 
-  enum Type { Normal, Password, Search, Calendar, Scan, CopyButton }
+  enum Type { Normal, Password, Search, Calendar, Scan, CopyButton, SendButton }
```
Comment (tomasMizera) [inline app/qmlV2/component/MMInput.qml:18]: Hi @iiLubos, I am afraid this is not a good way forward. We need to have standalone inputs for each of the types.

**Comment by jozefbudac** [comment]
Hoj, mor ho! detvo môjho rodu,
kto kradmou rukou siahne na tvoju slobodu:
a čo i tam dušu dáš v tom boji divokom:
Mor ty len, a voľ nebyť, ako byť otrokom.

File: app/qmlV2/component/MMInput.qml (hunk 1 above)
Comment (iiLubos) [reply app/qmlV2/component/MMInput.qml]: Refactored in this PR

File: gallery/notificationmodel.h [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+{
+    Q_GADGET
+
+  public:
+    Notification( uint id, const QString &message, uint interval, NotificationType::MessageType type, NotificationType::IconType icon );
+    uint id() { return mId; }
+    QString message() { return mMessage; }
+    NotificationType::MessageType type() { return mType; }
+    NotificationType::IconType icon() { return mIcon; }
+    bool canBeRemoved() { return ( mInterval-- == 0 ); }
```
Comment (tomasMizera) [inline gallery/notificationmodel.h:53]:
```suggestion
    bool isRemovableAfterDecrement() { return ( mInterval-- <= 0 ); }
```

**Review by tomasMizera — APPROVED** [review]

**Review by tomasMizera — APPROVED** [review]

**Pull request merged**